google/earthenterprise
# #607: Fix RPM Dependencies

* Moved `libpq` from the `opengee-server` to the `opengee-postgis` package.

* Added resolution of shell commands to packages that provide them for RPM (and Deb) packages.

* Added detection of provided and required capabilities using `rpm-build`'s `find-requires` and `find-provides`.

* Added support for `Requires(pre)` dependencies by rebuilding the RPM output by OS Package with `rpmrebuild`.

* Updated documentation.

## Verification Steps

* Build the OpenGEE RPMs.
    * Compile OpenGEE.
    * Stage install OpenGEE.
    * Package OpenGEE. (E.g.: `cd earthenterprise/earth_enterprise/rpms && ./gradlew openGeeRpms`)
* Transfer the RPMs to a machine with a minimum of packages installed.
* Enable the EPEL Yum repository.
* Install all RPMs in one command, such as `yum install opengee-*.rpm`.
* Verify the installation of all packages succeeds, and Server and Fusion start.  (You can make sure they pass a basic smoke test.)
* Uninstall OpenGEE RPMs. (E.g., `yum remove 'opengee-*'`.)
* Verify uninstallation succeeds.
